dask/dataframe/shuffle.py

```diff
@@ -302,6 +303,7 @@ def __init__(
         self.name_input = name_input
         self.meta_input = meta_input
         self.parts_out = parts_out or range(len(inputs))
+        super().__init__()
```

Comment:
This will give

```
TypeError: __init__() missing 7 required positional arguments: 'name', 'column', 'npartitions', 'npartitions_input', 'ignore_index', 'name_input', and 'meta_input'
```

Instead we want to call `__init__` for `Layer`

```suggestion
        Layer.__init__(self)
```